src/View/ViewBuilder.php

```diff
@@ -301,25 +302,61 @@ public function getPlugin(): ?string
     }
 
     /**
-     * Adds a helper to use.
+     * Adds a helper to use if no such helper with this name already exists.
      *
      * @param string $helper Helper to use.
      * @param array<string, mixed> $options Options.
      * @return $this
+     * @throws \Cake\Core\Exception\CakeException When a duplicate is found.
      * @since 4.1.0
      */
     public function addHelper(string $helper, array $options = [])
     {
-        $this->_helpers[$helper] = $options;
+        [$plugin, $name] = pluginSplit($helper);
+        if (isset($this->_helpers[$name])) {
+            $message = sprintf('The `%s` alias has already been added as helper. Use `setHelper()` instead.', $name);
+
+            throw new CakeException($message);
+        }
+
+        if ($plugin) {
+            $options['className'] = $helper;
```

Comment:
The same change should be done in `setHelper()` too. The array with class and config isn't exposed anywhere else and used only internally by ObjectRegistry.